<a href="https://colab.research.google.com/github/ekaratnida/DADS5001-Data-Analytics-and-Data-Science-Tools-and-Programming/blob/main/02_Filtering_%26_Sorting/Chipotle/Chipotle.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Ex1 - Filtering and Sorting Data




### Step 1. Import the necessary libraries

In [ ]:
import pandas as pd

### Step 2. Import the dataset from this [URL](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv).

### Step 3. Assign it to a variable called chipo.
show df.head()

In [ ]:
df = pd.read_csv("https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv",sep="\t")
df.head()

order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98

3.1 check dtypes of each column </br>
3.2 print describe </br>
3.3 df.shape

In [ ]:
print(df.dtypes)
print(df.describe())
print(df.shape)

order_id               int64
quantity               int64
item_name             object
choice_description    object
item_price            object
dtype: object
          order_id     quantity
count  4622.000000  4622.000000
mean    927.254868     1.075725
std     528.890796     0.410186
min       1.000000     1.000000
25%     477.250000     1.000000
50%     926.000000     1.000000
75%    1393.000000     1.000000
max    1834.000000    15.000000
(4622, 5)


In [ ]:
df.isnull().sum()

order_id                 0
quantity                 0
item_name                0
choice_description    1246
item_price               0
dtype: int64

### Step 4. Sort by the name of the item

In [ ]:
dfs = df.sort_values(by=['quantity','item_name'],ascending=False)
dfs.head()

order_id  quantity                     item_name choice_description  \
3598      1443        15  Chips and Fresh Tomato Salsa                NaN   
4152      1660        10                 Bottled Water                NaN   
3887      1559         8                 Side of Chips                NaN   
3599      1443         7                 Bottled Water                NaN   
2441       970         5                 Bottled Water                NaN   

     item_price  
3598    $44.25   
4152    $15.00   
3887    $13.52   
3599    $10.50   
2441     $7.50

In [ ]:
dfs.tail()

order_id  quantity          item_name choice_description item_price
4359      1741         1  6 Pack Soft Drink             [Coke]     $6.49 
4363      1742         1  6 Pack Soft Drink        [Diet Coke]     $6.49 
4522      1798         1  6 Pack Soft Drink        [Diet Coke]     $6.49 
4527      1800         1  6 Pack Soft Drink        [Diet Coke]     $6.49 
4535      1803         1  6 Pack Soft Drink         [Lemonade]     $6.49

### Step 5. What was the quantity of the most expensive item ordered?
Quantity = 15, (index = 1443)

In [ ]:
df.sort_values(by='item_price', ascending=False).head(1)

order_id  quantity         item_name  \
2624      1042         1  Steak Salad Bowl   

                                     choice_description item_price  
2624  [Fresh Tomato Salsa, [Black Beans, Sour Cream,...     $9.39

### Step 6. How many times was a Veggie Salad Bowl ordered?
ans = 18

In [ ]:
cs = df[df.item_name == 'Veggie Salad Bowl']
len(cs)

18

### Step 7. How many times did someone order more than one Canned Soda?
ans = 20

In [ ]:
step7 = df[(df.item_name == 'Canned Soda') & (df.quantity >1)]
step7

order_id  quantity    item_name choice_description item_price
18           9         2  Canned Soda           [Sprite]     $2.18 
51          23         2  Canned Soda     [Mountain Dew]     $2.18 
162         73         2  Canned Soda        [Diet Coke]     $2.18 
171         76         2  Canned Soda  [Diet Dr. Pepper]     $2.18 
350        150         2  Canned Soda        [Diet Coke]     $2.18 
352        151         2  Canned Soda        [Coca Cola]     $2.18 
698        287         2  Canned Soda        [Coca Cola]     $2.18 
700        288         2  Canned Soda        [Coca Cola]     $2.18 
909        376         2  Canned Soda     [Mountain Dew]     $2.18 
1091       450         2  Canned Soda       [Dr. Pepper]     $2.18 
1092       450         2  Canned Soda        [Coca Cola]     $2.18 
1944       787         2  Canned Soda       [Dr. Pepper]     $2.18 
2135       859         2  Canned Soda        [Diet Coke]     $2.18 
2235       901         4  Canned Soda           [Sprite]     $4.36 
2905      1156         2  Canned Soda        [Coca Cola]     $2.18 
2906      1156         2  Canned Soda           [Sprite]     $2.18 
3152      1258         2  Canned Soda       [Dr. Pepper]     $2.18 
3364      1349         2  Canned Soda        [Coca Cola]     $2.18 
3592      1440         2  Canned Soda        [Diet Coke]     $2.18 
3866      1550         2  Canned Soda     [Mountain Dew]     $2.18

### Step 8. How many products cost more than $10.00?
ans 31

In [ ]:
#prices = []
if df.item_price[0] == '$':
  prices = [float(value[1:-1]) for value in df.item_price]
df.item_price = prices
df2 = df.query('item_price > 10')
print(df2.item_name.nunique()) #remove duplicate item name
df2.shape

ValueError: Length of values (0) does not match length of index (4622)

### Step 9. What is the price of each item?

In [ ]:
df[(df.item_name == 'Chicken Bowl') & (df.quantity ==1)]

order_id  quantity     item_name  \
5            3         1  Chicken Bowl   
13           7         1  Chicken Bowl   
19          10         1  Chicken Bowl   
26          13         1  Chicken Bowl   
42          20         1  Chicken Bowl   
...        ...       ...           ...   
4590      1825         1  Chicken Bowl   
4591      1825         1  Chicken Bowl   
4595      1826         1  Chicken Bowl   
4599      1827         1  Chicken Bowl   
4604      1828         1  Chicken Bowl   

                                     choice_description item_price  
5     [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
13    [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...    $11.25   
19    [Tomatillo Red Chili Salsa, [Fajita Vegetables...     $8.75   
26    [Roasted Chili Corn Salsa (Medium), [Pinto Bea...     $8.49   
42    [Roasted Chili Corn Salsa, [Rice, Black Beans,...    $11.25   
...                                                 ...        ...  
4590  [Roasted Chili Corn Salsa, [Rice, Black Beans,...    $11.25   
4591  [Tomatillo Red Chili Salsa, [Rice, Black Beans...     $8.75   
4595  [Tomatillo Green Chili Salsa, [Rice, Black Bea...     $8.75   
4599      [Roasted Chili Corn Salsa, [Cheese, Lettuce]]     $8.75   
4604  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...     $8.75   

[693 rows x 5 columns]